dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java

```diff
@@ -682,10 +687,16 @@ private DependResult isTaskDepsComplete(String taskName) {
                 return DependResult.WAITING;
             }
             ExecutionStatus depTaskState = completeTaskList.get(depsNode).getState();
-            // conditions task would not return failed.
-            if(depTaskState.typeIsFailure()
-                    && !DagHelper.haveConditionsAfterNode(depsNode, dag )
-                    && !dag.getNode(depsNode).isConditionsTask()){
+            // conditions task should be handled separately
+            if (dag.getNode(depsNode).isConditionsTask()) {
+                List<String> tmpTaskList = parseConditionTask(depsNode);
+                if (tmpTaskList.contains(taskName)){
```

Comment:
you must scan all the depend nodes before return success.

Reply:
In fact, in the loop it is supposed to return FAILED when dependency is not satisfied. And if all dependencies are satisfied it will **return SUCCESS outside the loop.**